napari/layers/points/points.py

```diff
@@ -1465,10 +1465,15 @@ def _slice_data(
         """
         # Get a list of the data for the points in this slice
         not_disp = list(self._dims_not_displayed)
-        indices = np.array(dims_indices)
+        # We want a numpy array so we can use fancy indexing with the non-displayed
+        # indices, but as dims_indices can (and often/always does) contain slice
+        # objects, the array has dtype=object which is then very slow for the
+        # arithmetic below. As Points._round_index is always False, we can safely
+        # convert to float to get a major performance improvement.
+        not_disp_indices = np.array(dims_indices)[not_disp].astype(float)
```

Comment:
Actually, maybe something like the following is even safer (just in case `dims_indices` actually contains ints that would be lossily convert to floats`) and more readable?

```python
not_disp_indices = np.array([dims_indices[i] for i in not_disp])
```